File: ICOBackEnd/src/main/java/com/example/icobackend/models/Vehicle.java
package com.example.icobackend.models;

import lombok.Data;

@Data
public class Vehicle {

    private int capacity;
    private double costPerDistance;

    public Vehicle(int capacity, double costPerDistance) {
        this.capacity = capacity;
        this.costPerDistance = costPerDistance;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }

    public double getCostPerDistance() {
        return costPerDistance;
    }

    public void setCostPerDistance(double costPerDistance) {
        this.costPerDistance = costPerDistance;
    }
}

File: icoFrontEnd/src/app/models/Vehicle.ts
import { Coordinate } from "./Coordinate";

export class Vehicle {
    
    capacity: number;
    costPerDistance: number;
    route!: Coordinate[];

    constructor(capacity: number, costPerDistance: number) {
        this.capacity = capacity;
        this.costPerDistance = costPerDistance;
    }
}